ast/env.go

```diff
@@ -253,6 +270,12 @@ func (n *typeTreeNode) Put(path Ref, tpe types.Type) {
 	curr.value = tpe
 }
 
+func (n *typeTreeNode) PutArg(path Ref, i int, tpe types.Type) {
+	pos := json.Number(strconv.Itoa(i))
```

Comment:
And it can't be ambiguous, all overloaded functions have to share the same number of arguments.